### DISCUSSION TITLE
Using entity specific custom features in NER

### DISCUSSION THREAD

**Discussion by mapadofu** [explosion/spaCy#10894] [Help: Coding & Implementations]
Generically, is there a way to use additional/custom features as inputs to NER model?  I vaguely recall that earlier versions of the software did support this  and how to do it was described in the documentation.  However I haven't been able to locate this in the current version.

Specifically, I've got some custom entity types similar to serial numbers.  I can flag most instances of them using a rule-based matcher, and could use that matcher to set custom attributes on the token that would indicate "this token 'looks like' a serial number".  The idea would be to incorporate these highly informative features into the NER model that I plan on training on the relevant dataset.

**Comment by polm** [comment]
What you may be remembering is that you can customize the attributes used in [the `tok2vec` layer](https://spacy.io/api/architectures#MultiHashEmbed). This can capture information like prefixes/suffixes, whether a token is numeric, etc. 

The default tok2vec only supports Token attributes (the ones [here](https://github.com/explosion/spaCy/blob/85dd2b6c045854c4e07dc5daa40dfbfd54043252/spacy/attrs.pyx#L5)). You can implement a custom tok2vec layer to use arbitrary information, though it's a little more complicated, and will be slower. You can see an example of someone doing that [here](https://github.com/explosion/spaCy/discussions/6527#discussioncomment-186516)).
